### PR TITLE
Try to build 0.7rc1 of SunPy

### PR DESCRIPTION
**Pull request by Cadair**

### CODE REVIEW

**Comment by Cadair** [comment]
@mwcraig any way to upload to a different label for pre-release versions?!

**Comment by mwcraig** [comment]
Not at the moment, but (relatively) easy to implement, I think. Does logic something like this work?

```
if letters_are_in_version_number:
    upload_package_to_alternate_label
else:
    upload_package_to_main_label
```

Also, do you see a need for separate labels for dev releases (e.g. `0.7.dev1234`) and release candidates/alpha/beta (e.g. `0.7.rc1` or `0.7.beta1`)?

**Comment by mwcraig** [comment]
Builds are failing because it can't download the course tarball...

**Comment by Cadair** [comment]
@mwcraig yes maybe that's because pypi put dev versions somewhere else?!

**Comment by mwcraig** [comment]
@Cadair -- more likely the fact pypi.io is still in development. Can you try changing the url to something like this:

`https://pypi.python.org/packages/source/s/sunpy/sunpy-{{version}}.tar.gz`

**Comment by Cadair** [comment]
This URL does not work either: https://pypi.python.org/packages/source/s/sunpy/sunpy-0.7rc1.tar.gz

**Comment by mwcraig** [comment]
I really hope this isn't a permanent change in how one resolves links to pypi packages...

**Comment by mwcraig** [comment]
I did, incidentally, get Windows 64-bit building last night, still need to merge the change, though. A couple more changes to make that I'll do that now, and then, if you want, you can remove the skip on Windows.

**Comment by Cadair** [comment]
the py35 windows skip is because of glymur.

**Comment by mwcraig** [comment]
Windows build is going to fail on `import sunpy` again, so canceling appveyor. I'll ping you to rebase in a bit, and then Windows should succeed.

**Comment by mwcraig** [comment]
@Cadair -- can you please rebase?

**Comment by Cadair** [comment]
@mwcraig done

**Comment by mwcraig** [comment]
@Cadair -- looking into the source of the latest fails. Could you do me a favor and also push your branch to this repo (openastronomy/conda-channel)?

That would help determine whether the problem really is with not having a `BINSTAR_TOKEN` set, as I suspect.

In the meantime I'm restarting the appveyor build in hope it was just an intermittent thing a anaconda.org.

File: recipe_templates/sunpy/meta.yaml [hunk 1]
```diff
@@ -4,13 +4,13 @@ package:
...
   fn: sunpy-{{version}}.tar.gz
-  url: https://pypi.io/packages/source/s/sunpy/sunpy-{{ version }}.tar.gz 
+  url: https://pypi.python.org/packages/75/d3/9b5578d40feaa7adb535b4ba40a455404989482f4fca1b2bf18e4bdb3ca9/sunpy-{{ version }}.tar.gz
   md5: {{ md5 }}
 
 build:
   number: 0
   script: python setup.py install --offline --no-git --single-version-externally-managed --record record.txt
-  skip: True  # [py>27]
+  skip: True  # [win and py35]
```
Comment (mwcraig) [inline recipe_templates/sunpy/meta.yaml:13]: I think this should be `[win and py3k]`  to catch both python 3.4 and 3.5.

Comment (Cadair) [inline recipe_templates/sunpy/meta.yaml:13]: the build should work under 3.4 as there should be glymur packages?

**Comment by Cadair** [comment]
pushed to a branch.

File: recipe_templates/sunpy/meta.yaml (hunk 1 above)
Comment (mwcraig) [inline recipe_templates/sunpy/meta.yaml:13]: could be...I have too many packages I'm keeping track of right now.  Just looked, and you are right, it is _building_ sunpy successfully on Windows/python 3.4, but it fails when it tries to test the build. Not sure why, let's see what happens on the branch build

**Comment by Cadair** [comment]
say waaaa https://ci.appveyor.com/project/openastronomy/conda-channel/build/1.0.56/job/06lq94t0o70p0cyu#L2037

**Comment by mwcraig** [comment]
> say waaaa 

yeah....that. Oddly, that is what was happening when I used an outdated version of conda-build in python 2.7 locally. Updating conda-build seemed to fix it (on 2.7). 

Will try to reproduce locally, but could you also try altering https://github.com/OpenAstronomy/conda-channel/blob/sunpy-07rc1/appveyor.yml#L81 to use conda-build 1.20.3 just in case that fixes?

It looks like there is a separate issue with conda-build-all in cases when a valid binstar token is not defined, but we can work around that for now by using branches in the  openastronomy repo.

**Comment by mwcraig** [comment]
I have no idea what the underlying issue is. When I build locally on python 3.4/win64 I don't see the error. I can, at least for now, upload my local build if you'd like.

**Comment by mwcraig** [comment]
For reference, this discussion of handling pre-release uploads: https://github.com/astropy/conda-build-tools/issues/13

**Comment by bsipocz** [comment]
This is probably need to be closed as there are newer releases of sunpy in this channel.

**Pull request closed without merging**